pure.zsh

```diff
@@ -470,11 +472,25 @@ prompt_pure_setup() {
 	add-zsh-hook precmd prompt_pure_precmd
 	add-zsh-hook preexec prompt_pure_preexec
 
+	local ssh_connection=$SSH_CONNECTION
+	local username
+	if [[ -z $ssh_connection ]] && (( $+commands[who] )); then
+		# When changing user on a remote system, the $SSH_CONNECTION
+		# environment variable can be lost, attempt detection via who.
+		if who am i | grep -q '(.*)$' &>/dev/null; then
+			ssh_connection=true
+		fi
+	fi
+
 	# show username@host if logged in through SSH
-	[[ "$SSH_CONNECTION" != '' ]] && prompt_pure_username='%F{242}%n@%m%f'
+	[[ -n $ssh_connection ]] && username='%F{242}%n@%m%f'
 
 	# show username@host if root, with username in white
-	[[ $UID -eq 0 ]] && prompt_pure_username='%F{white}%n%f%F{242}@%m%f'
+	[[ $UID -eq 0 ]] && username='%F{white}%n%f%F{242}@%m%f'
+
+	typeset -gA prompt_pure_state=(
+		username "$username"
+	)
```

Comment:
I'm considering doing more checks at "startup" in the future, the result of those would also go into `prompt_pure_state`. We should be able to get rid of a bunch of other global (internal) pure variables this way.

Reply:
I like it 👍